flowapi/flowapi/permissions.py

```diff
@@ -1,245 +1,64 @@
-    paths: Optional[List[str]] = None,
-    argument_names_to_extract: List[str] = ["aggregation_unit"],
-) -> Tuple[List[str], dict]:
-    """
-    Yield the paths to the leaves of a tree and the associated leaf node.
-
-    Parameters
-    ----------
-    *
-    paths : list of str
-        Parents of this path
-    tree : dict
-        Tree of queries
-
-    Yields
-    ------
-    Tuple of list, dict
-    """
-    if paths is None:
-        paths = []
-    new_path = list(paths)
-    if "properties" in tree and "query_kind" in tree["properties"]:
-        new_path = paths + tree["properties"]["query_kind"]["enum"]
-    if "oneOf" in tree.keys():  # Path divergence
-        for tr in tree["oneOf"]:
-            yield from enum_paths(paths=new_path, tree=tr)
-    elif (
-        "enum" in tree.keys()
-        and len(tree["enum"]) > 1
-        and new_path[-1] in argument_names_to_extract
-    ):
-        yield (new_path, f"{{{new_path[-1]}}}")
-    elif "items" in tree.keys():
-        yield from enum_paths(paths=new_path, tree=tree["items"])
-    else:
-        if "properties" in tree:
-            for k, v in tree["properties"].items():
-                if k == "query_kind":
-                    yield new_path,
-                else:
-                    yield from enum_paths(paths=new_path + [k], tree=v)
-
-
-def paths_to_nested_dict(
-    *, queries: dict, argument_names_to_extract: List[str] = ["aggregation_unit"]
-) -> dict:
-    d = {}
-    for x in list(
-        enum_paths(tree=queries, argument_names_to_extract=argument_names_to_extract)
-    ):
-        path, *key = x
-        the_d = d
-        for v in path[:-1]:
-            the_d = the_d.setdefault(v, dict())
-        if len(key) > 0:
-            the_d.setdefault(path[-1], list()).append(key[0])
-        else:
-            the_d.setdefault(path[-1], dict())
-    return d
-
-
-def dict_and_list_key_function(
-    x: Tuple[str, Union[dict, list, tuple]],
-) -> Union[Tuple[bool, int, str], Tuple[bool, str]]:
+def is_flat(in_iter):
     """
-    Key function which sorts itemviews. Sorts items where the value is
-    a list or a dict _after_ those where it isn't, then sorts dicts and
-    lists by their length, then by the key.
-
-
-    Parameters
-    ----------
-    x : tuple
-        Item to get a key for.
-
-    Returns
-    -------
-    tuple of bool, int and input or tuple of bool and input
-        False, negative length of input and then key for dicts and lists
-        True, key for other tuples
-
+    Returns True if in_iter is flat (contains no dicts or lists)
     """
-    if isinstance(x[1], (dict, list)):
-        return False, -len(x[1]), x[0]
-    else:
-        return True, x[0]
+    if not isinstance(in_iter, collections.Container):
+        return True
+    if isinstance(in_iter, dict):
+        in_iter = in_iter.values()
+    # Think there's a slightly better way of doing type introspection here
+    return all(type(item) not in [dict, list] for item in in_iter)
```

Comment:
Could do `not any(isinstance(item, (dict, list)) for item in in_iter)` I guess? Not sure that's more satisfying really.

Reply:
My other thought was `isinstance(Collection)` or similar, but something went wrong with that during dev.